### DISCUSSION TITLE
Connection failure

### DISCUSSION THREAD

**Discussion by Lynd** [NightscoutFoundation/xDrip#3004] [General]
I started a new G7 sensor 2 days ago and it worked fine until today.  A few hours ago, I stopped receiving readings and saw that Connection Status showed "not connected".    In desperation, I turned off my Android bluetooth and then turned back on and then re-entered the Dexcom Transmitter ID in hopes that it would reset the connection.   Now, the Scan Error" on the System Status page displays "Enable bluetooth and try again".

**Comment by Navid200** [comment]
Are all settings correct?
https://navid200.github.io/xDrip/docs/G6-Recommended-Settings.html

**Comment by Lynd** [comment]
After checking the settings, a few were set differently so I corrected those and restarted the phone.  Now, the status has been stuck on "hunting transmitter" for a while.

**Reply by Navid200** [reply]
Was one of the wrong settings having "Allow OB1 unbonding" enabled?

Can you tell us what the phone brand and model is?

**Comment by Lynd** [comment]
"Was one of the wrong settings having "Allow OB1 unbonding" enabled?"
I don't believe so
"Can you tell us what the phone brand and model is?"
Samsung S21 Ultra

**Reply by Navid200** [reply]
Can you post screenshots of the G5/G6/Dex1 status page as well as the logs?

**Reply by Lynd** [reply]
Sure. Would you like me to screenshot the most current page of the logs or more?

**Comment by Lynd** [comment]
I also just noticed that the Sensor Start date shows over 12 days ago, which would have been the start time of the the sensor before this one.

**Reply by Navid200** [reply]
Are the pairing codes the same between the current device and the one you had before?

**Reply by Lynd** [reply]
I don't know how to check since I no longer have the sensor boxes. Is the pairing number contained in the logs?   

I also see that the firmware version indicates "32.192.104.109 Unknown"

**Reply by Navid200** [reply]
I need to see the status page.

**Reply by Lynd** [reply]
![temp5](https://github.com/NightscoutFoundation/xDrip/assets/8272764/b9ec7566-8743-4355-9cb8-f7b17364c0f0)

**Reply by Navid200** [reply]
And you have blacked out the version of xDrip in that screenshot to accomplish what?!

**Comment by Lynd** [comment]
Sorry. I thought that was some kind of unique ID that shouldn't be posted on the Internet. Here it is without redaction. 
![temp4](https://github.com/NightscoutFoundation/xDrip/assets/8272764/b7bc6039-3a87-4f07-8a0d-b44799117302)

**Reply by Navid200** [reply]
If you look at the release notes, you can see that we have added a new G7 firmware at a release newer than the one you are using.
https://navid200.github.io/xDrip/docs/ReleaseNotes.html

So, the first thing you should do is to update your xDrip to the current stable release.

**Comment by Lynd** [comment]
Here's the updated status page

![temp5](https://github.com/NightscoutFoundation/xDrip/assets/8272764/91213621-d384-4702-8653-3036f8570bb5)

**Reply by Navid200** [reply]
Those battery voltages are low.
How old is this device?  Did you receive it a long time ago?

**Reply by Navid200** [reply]
If you go to the logs and enable all categories and search for volt, you will see the battery voltages over the past few days.  Do you want to do that and post a screenshot?

**Reply by Lynd** [reply]
![temp4](https://github.com/NightscoutFoundation/xDrip/assets/8272764/9982705f-90a8-47f7-8208-d6ac28132f4e)

**Reply by Navid200** [reply]
Did you install the Dexcom app today by any chance?  Or, did you install xDrip on another phone and used it to control the same device?

**Reply by Lynd** [reply]
No to both questions.

**Reply by Navid200** [reply]
248 is the lowest I have seen for the G7 voltage B.  But, I cannot tell you that I am sure that is the cause.

Your previous device was higher.  But, still low.  It was at 255 and gradually dropped to 252.

If you want, you can use a scanner to see if you can see the transmitter ping?
https://navid200.github.io/xDrip/docs/Bluetooth-Scanner.html

**Reply by Lynd** [reply]
With the scanner, DXCMPF device shows up with a -64 reading.   The connect button is grayed out.

**Reply by Navid200** [reply]
Please go to previously connected Bluetooth devices list on your phone.
Can you show what you see?

**Reply by Lynd** [reply]
Not sure whether this is the screen you want to see, but here is the screen for this device:

![temp4](https://github.com/NightscoutFoundation/xDrip/assets/8272764/c0075301-cbb8-4f52-89a2-e0a0617c754f)

**Reply by Navid200** [reply]
Is it possible that DXCMPf is actually your previous device?
Where is it?  Is it in the trash still inside the house?

**Reply by Lynd** [reply]
I'm not 100 certain where it is. Can I just go outside and re-scan to see if it disappears?

**Reply by Navid200** [reply]
Go to logs, enable all categories and search for DXCM.  What do you see?

**Reply by Lynd** [reply]
Searching for DXCM produces no results.

**Reply by Navid200** [reply]
OK

Different question:  Are you sure 2257 is the correct pairing code for the current device?

**Reply by Lynd** [reply]
I think so, but I'm not certain now that so many changes have been made. Should that be in the logs?

**Reply by Navid200** [reply]
What do you get if you search for transmitter in the logs?

**Reply by Lynd** [reply]
![temp6](https://github.com/NightscoutFoundation/xDrip/assets/8272764/b245c6f4-215a-4c7e-9e17-4f2d5f85d7d0)
![temp7](https://github.com/NightscoutFoundation/xDrip/assets/8272764/e28cf694-b24e-4f54-94fc-2aba40679d33)

**Reply by Lynd** [reply]
Is it possible to search the logs for entries near a certain time/day?

**Reply by Navid200** [reply]
Good point.
The logs are shown in a chronological order.  So, by scrolling through them, you can view the logs associated with different time periods.

**Reply by Lynd** [reply]
There are so many logs entries that it's difficult to locate things by scrolling, but I don't see much around the 8-10 22:15 time.  Is there anything else I can try to find the transmitter ID?  Otherwise, it seems like I'll have to remove this transmitter and start a new one.

**Reply by Navid200** [reply]
Is the applicator gone?  Is it not somewhere you can find?

**Reply by Lynd** [reply]
I can't find it.

**Reply by Navid200** [reply]
I'm sorry I don't have a hands-on experience with G7 myself yet.
And the developer who does is away.

**Reply by Lynd** [reply]
I understand. BTW, I believe I have the right transmitter ID because if I change it to anything else, the Voltage indicators disappear. When I change it back to the correct ID, the voltage indicators re-appear.     Should I try removing it from my Android bluetooth settings?

**Reply by Navid200** [reply]
This proves that the ID corresponds to a device you have used before.  it does not prove that it is necessarily the current device.  
xDrip has a memory of all the devices you have used.  
So, it could be the current device.  or, it could be the previous device.

**Reply by Navid200** [reply]
You said that the scanner showed DXCMPf. 
Where exactly is your previous device?

**Reply by Lynd** [reply]
I wouldn't have entered the previous device ID in the last 12 days.  So unless xDrip automatically reverted to an old ID after the current ID/sensor was working, I think the current transmitter ID is the one being used.

**Reply by Navid200** [reply]
In that case, I don't understand why you are not getting a pair request if all the settings are correct.

**Reply by Lynd** [reply]
Thanks for trying.  Do you know when the developer with G7 experience will be back?  I'm apprehensive about trying another sensor in case the same thing happens.

**Reply by Navid200** [reply]
I'm afraid it will be the end of the week.

**Reply by Navid200** [reply]
Can you not use the Dexcom app?

**Reply by Lynd** [reply]
I did try the sensor on the Dexcom app, and it worked fine. Unfortunately I had already started a new sensor on xDrip, so now I have two working sensors.  Hopefully when your colleague is back he will comment on the issue.

**Reply by Navid200** [reply]
Are you saying the device we were trying to bring up all along is now working?

**Reply by Lynd** [reply]
Yes--only on the Dexcom app.

**Comment by Lynd** [comment]
The G7 device?  It's only a couple of weeks old.

**Comment by Lynd** [comment]
Hi. I'm just following up on this to see whether the developer with G7 experience would comment.  My G7 sensor continued to work with the the Dexcom app, but would not with xDrip.   

Now I have a new G7 sensor working with xDrip, but I'm concerned that I'm inevitably going to experience the same pairing failure.

**Reply by jamorham** [reply]
To diagnose anything like this further, the first thing you'd need to do is enable OB1 debugging: https://github.com/NightscoutFoundation/xDrip/wiki/Enabling-Debugging-for-OB1-collector and make sure you are running the latest xDrip nightly and then try to use the sensor and if you get problems then use the log upload feature and we can take a look and the details.
